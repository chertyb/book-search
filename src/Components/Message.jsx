import React from 'react';
import PropTypes from 'prop-types';

const Message = (props) => {
  const { message, type } = props;
  return <p className={type}>{message}</p>;
};

Message.propTypes = {
  message: PropTypes.string,
  type: PropTypes.string,
};

Message.defaultProps = {
  message: '',
  type: '',
};

export default Message;
