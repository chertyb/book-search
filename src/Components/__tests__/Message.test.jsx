import React from 'react';
import ReactDom from 'react-dom';
import { shallow } from '../../enzyme';
import Message from '../Message';

describe('Message component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDom.render(<Message />, div);
    expect(div).toMatchSnapshot();
    ReactDom.unmountComponentAtNode(div);
  });

  it('should display props value', () => {
    const message = 'Error msg';
    const wrapper = shallow(<Message message={message} />);
    expect(wrapper.text()).toEqual(message);
  });
});
