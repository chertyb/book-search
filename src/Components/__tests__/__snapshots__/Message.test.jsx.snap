// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component renders without crashing 1`] = `
<div>
  <p />
</div>
`;
